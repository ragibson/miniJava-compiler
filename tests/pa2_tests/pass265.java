//  PA2 pass expr precedence
class A {
    int f (){
        int d = 2+-x- -x;
    }
}
