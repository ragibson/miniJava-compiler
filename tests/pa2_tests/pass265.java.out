********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   INT BaseType
  .   .   "f" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [1]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "d" varname
  .   .   .   BinaryExpr
  .   .   .     "-" Operator
  .   .   .       BinaryExpr
  .   .   .         "+" Operator
  .   .   .           LiteralExpr
  .   .   .             "2" IntLiteral
  .   .   .           UnaryExpr
  .   .   .             "-" Operator
  .   .   .               RefExpr
  .   .   .                 IdRef
  .   .   .                   "x" Identifier
  .   .   .       UnaryExpr
  .   .   .         "-" Operator
  .   .   .           RefExpr
  .   .   .             IdRef
  .   .   .               "x" Identifier
=============================================
